name-registry/src/interface.sw

```diff
@@ -0,0 +1,99 @@
+library interface;
+
+abi NameRegistry {
+    /// Extends the duration of ownership for the name
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The name to extend the duration of
+    /// * `duration` - The duration to extend by
+    ///
+    /// # Reverts
+    ///
+    /// * If the name is not registered
+    /// * If the payment is not sufficient for the duration
+    /// * If the wrong asset is sent
+    #[storage(read, write)]
+    fn extend(name: str[8], duration: u64);
+
+    /// Registers an entry into the registry for the given name, assigns it to the given owner, and resolves the name to the given identity when queried
```

Comment:
This is too verbose for a summary. It should be a single overview stating that a name is registered and then after an empty new line, on the 3rd line, it can contain additional information about the resolution etc.